### DISCUSSION TITLE
Do Zed aims to become alternative for VSCode or Webstorm?

### DISCUSSION THREAD

**Discussion by keinsell** [zed-industries/zed#6575] [Feature Requests]
I wonder where's the vision of project founders, I really loved Atom which was my first editor and Zed is probably continuation but honestly I do not think building VSCode alternative is a viable option on market - especially when Zed have powerful multi-threading features which is something that for example is missing in `Webstorm`.

**Comment by widersky** [comment]
I hope so. VSCode is ugly (and heavy) as hell and it's based on Electron which basically means "VSCode need separate Chromium instance just to run" - which is crazy from my point of view. On the other side - WebStorm and related IDEs are heavy as hell too, but for another reason - they're based on Java. I have never needed anything new and light in this field like I do now in my career. We have Sublime Text, but transparency (or rather lack thereof) in communication and lack of clarity about the direction of development repels me from this editor.

I would use Zed every day, but it's lacking PHP and Dart support (which will probably appear as soon as we get the option to expand the editor through plugins). I also miss folder icons and a slightly clearer project view. Besides, it is already an almost perfect editor.

**Reply by keinsell** [reply]
Actually I'm kinda pissed about everybody building VS Code alternatives, like we have enough "lightweight" editors on market from Vim, Fleet, VSC, Nova, Cursor and a lot of other crap that do not introduce anything new and lacks fundamental ecosystem to be useful. However, we have zero alternatives to InteliJ which as you said are fucking heavy.

Sadly Zed do not deliver anything beside Copilot integration and Language Server which are things that Neovim can provide (both option are performant) - I would rather say until there will be good toolkit for ex. refactorings and code sense it's kinda another VSCode but without extensions.

**Reply by widersky** [reply]
What you write about is true. Until there are extensions and the entire ecosystem around Zed, it will not be a complete experience. There are actually a lot of editors and I have tested many of them, but everyone has their own problems - often resulting literally from the architecture behind them. Nova, for example, does not support some Language Server Protocol instructions, making it impossible to sometimes add critical functionalities to extensions. Besides, the creators of Nova do not care too much about communication so you don't really know what they are working on and when to expect an update. When paying for an editor, I would expect full transparency - and yes, I will not renew my license in a year. Additionally, Nova seems to be an less and less stable editor with each update, even untrustworthy.
Fleet from JetBrains is a dead end. The editor was supposed to be light, and doing nothing takes up 1GB of RAM for me, and when I turn on Smart mode, the value rises well above what IntelliJ takes! Total misunderstanding.
That's why I strongly believe in Zed and hope to take some VSCode cake, because I'm just tired of the widespread admiration for VSCode in some communities and pushing it literally everywhere as if it were a golden grail.

**Reply by mocenigo** [reply]
> What you write about is true. Until there are extensions and the entire ecosystem around Zed, it will not be a complete experience. 

Well, there *are* extension, the ecosystems needs to mature, still, but one can write extensions.
Zed does not have VSCode extensions compatibility, but we do not want to add an entire browser inside it, right?

**Reply by keinsell** [reply]
> Zed does not have VSCode extensions compatibility, but we do not want to add an entire browser inside it, right?

I agree.

**Comment by albas99** [comment]
Currently I think the only lightweight alternative to VSCode that has the support and ecosystem to back it up is Nvim, all Jetbrains products in my opinion are built for the developer with a high-end machine that can handle their size and memory hog. VSCode wants to be that editor that everyone can use, the independent developer with a mid-level computer, the beginner with a relatively low-end laptop learning to code and the professional developer but it's foundation being electron-based automatically contradicts that(at least for devs with limited capability computers). Zed is new and I'm optimistic that they'll introduce more features in the future but until then, especially an extension ecosystem, I don't think people will find it compelling enough to switch completely.

**Comment by thinisde** [comment]
In my opinion, zed is now the most potential IDE in 2024. Now it is open source and developers can have access to it, Zed will explode so much more. People will create so many packages to Zed and it will replace VSCode. 

VSCode is really trash right now, as you said, it is based on electron and chromium. The collaboration feature has a lot of potential, it has built in chat-gpt and copilot. I would use Zed as my main IDE in the future, but currently it still lack a lot of support, e.g. community packages, .java syntax support and other stuff.

**Reply by keinsell** [reply]
Do not use word "IDE" until you are able to export call graph of function and have solid debugger experience 

Its text editor at is's best.

**Reply by endurance-chorus** [reply]
When Zed moves from text editor -> IDE, I'll never look back. Webstorm is getting slower every year, and its unacceptable.

**Reply by mocenigo** [reply]
> When Zed moves from text editor -> IDE, I'll never look back. Webstorm is getting slower every year, and its unacceptable.

I hope it stays as text editor with IDE functionality, but with better support for "isolated" files.

**Comment by kb9c4nprcb-art** [comment]
Zed could clean up if they could somehow move into Jetbrains territory.  Jetbrains is seemingly rudderless and completely out of touch with what their customers want (#1 ask: editor open files <10s and w/o crashing).  I've been a paying customer for years (of Jetbrains) but I feel like I have no option but to cancel when I can't use their product because it's so slow or crashes because it can't handle so many out-of-band fs changes, from coding agents, or whatever the hell is causing it.  I'm on a fully stock IntelliJ config, on an M2 Max w/ 96gb ram.  I should not be waiting for my editor to open a file.

Here's what Zed would need to do that:
- Debugger API obviously
- Stable and improved plugin ecosystem
- Learn from Jetbrains mistakes.  Do not push your AI subscription.  My opinion is: be the editor that is most integrated with everyone else's subscription.  You'll claw my Claude Max 20 from my cold hands.  But if using CC is the best inside Zed?  Sign me up!
- Jetbrains keymap
- A great merge tool
- Don't fuck up the good things about Zed

Years ago Jetbrains made a bet, at a time when everyone thought the days of paying for an IDE were long gone.  Everyone used Eclipse and everyone was pretty meh about it.  Jetbrains thought that people would pay if they could make something better.  And they just kept improving it year after year, slowly adding the right features and not getting distracted.  And today, no one would dream of using Eclipse.  It's VSCode with the massive install base, and Jetbrains as the only real competitor.  

But Jetbrains has lost their way it seems and there is a great opportunity for someone else to come along and build the new curated excellent text editor/IDE for the agentic age.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
